### DISCUSSION TITLE
how can I update job date?

### DISCUSSION THREAD

**Discussion by alyatwa** [timgit/pg-boss#398] [Q&A]
if I want to update startAfter date currently I cancel running job and send it again to the queue, is there better way to do this?

```js
        const jobQuery = `SELECT * FROM "pgboss"."job"  WHERE "singletonkey"='onEventEndedQueue-${event.id}';`
        const result: any = await pgClient.query(jobQuery)
        if (result.rows.length > 0) {
          let jobs = result.rows.map((a: JobType) => a.id)
         await boss.cancel(jobs)
        }
//////////////
await boss.sendAfter(
        onEventEndedQueue,
        { eventId: event.id },
        {
          singletonKey: `${onEventEndedQueue}-${event.id}`,
        },
        event.endDate
      )
```

**Comment by Rana-Almihdhar** [comment]
>  if I want to update startAfter date currently I cancel running job and send it again to the queue, is there better way to do this?
> 
> 
> 
> 
> 
> ```js
> 
>         const jobQuery = `SELECT * FROM "pgboss"."job"  WHERE "singletonkey"='onEventEndedQueue-${event.id}';`
> 
>         const result: any = await pgClient.query(jobQuery)
> 
>         if (result.rows.length > 0) {
> 
>           let jobs = result.rows.map((a: JobType) => a.id)
> 
>          await boss.cancel(jobs)
> 
>         }
> 
> //////////////
> 
> await boss.sendAfter(
> 
>         onEventEndedQueue,
> 
>         { eventId: event.id },
> 
>         {
> 
>           singletonKey: `${onEventEndedQueue}-${event.id}`,
> 
>         },
> 
>         event.endDate
> 
>       )
> 
> ```

**Comment by timgit** [comment]
You could also use an UPDATE command in SQL

**Reply by Rana-Almihdhar** [reply]
but why when I cancel a job it also got executed

**Reply by leland-86** [reply]
Is there any way to do this that doesn't rely on us using raw SQL in the application layer? I would expect the API to be able to abstract that logic by just passing in an id without having to delete and insert

**Comment by alyatwa** [comment]
but why when I cancel a job it also got executed

**Reply by timgit** [reply]
Cancelled jobs cannot be fetched using work() or fetch().
